MapboxDirections/MBVisualInstructionComponent.swift

```diff
@@ -43,11 +49,22 @@ public class VisualInstructionComponent: NSObject, NSSecureCoding {
             #else
                 scale = UIScreen.main.scale
             #endif
-            self.imageURL = URL(string: "\(baseURL)@\(Int(scale))x.png")
+            imageURL = URL(string: "\(baseURL)@\(Int(scale))x.png")
         }
+        
+        self.init(text: text, imageURL: imageURL)
     }
     
-    public required init?(coder decoder: NSCoder) {
+    /**
+     :nodoc:
+     Initialize A `VisualInstructionComponent`.
+     */
+    @objc public init(text: String?, imageURL: URL?) {
```

Comment:
In the future, as we add additional fields to the component object, if we find that the number of arguments gets unwieldy, we can make the properties `var` instead and leave them out of the initializer.